src/types/InstantEvent.ts

```diff
@@ -0,0 +1,23 @@
+import {GeneralEvent} from './GeneralEvent';
+//Instant Event type
+export class InstantEvent implements GeneralEvent<string, string>{
+    name: string;
+    cat: string;
+    readonly ph: string = "i";
+    ts: number;
+    tts?: number;
+    pid: number;
+    tid: number;
+    args: Map<string, string> = new Map();
+    cname?: string;
+    s: string; //extra parameter (default value: "t")
+
+    constructor(name: string, cat: string, ts: number, pid: number, tid: number, s: string){
```

Comment:
You can add the public modifier here and you won't have to instantiate the properties mentioned as arguments for the constructor

So essentially 

```constructor(public name: string, public cat:string, public ts:number, public pid:number, public tid: number, public s:string)``` and then you can remove these in the body of the constructor as well

Reply:
Got it, close this and push on a new PR now as discussed!